components/scream/src/share/tests/vertical_interp_tests.cpp

```diff
@@ -11,46 +11,192 @@ using Spack = SmallPack<Real>;
+  //the output levels are different from the input levels:
+  //1) n_layers_src= 2*N,   n_layers_tgt= 2*N
+  //2) n_layers_src= 2*N+1, n_layers_tgt= 2*N
+  //3) n_layers_src= 2*N,   n_layers_tgt= 2*N-1
+  //4) n_layers_src= 2*N+1, n_layers_tgt= 2*N-1
+  //For each scenario target levels are at the midpoint and so should be the average
+  //of the source layers. 
 
-  const int n_layers_src = 3;
-  const int n_layers_tgt = 4;
+  int n_layers_src[4] = {2*N,2*N+1,2*N,2*N+1};
+  int n_layers_tgt[4] = {2*N,2*N,2*N-1,2*N-1};
+
+  printf (" -- Testing vertical interpolation with Pack size %d --\n",N);
+
+  for (double perturb : {0, 1}){
+    printf ("    -> Target pressure levels: p_tgt = p_src + %f\n",perturb);
+    for (int i=0; i<4; i++){
+      if (perturb == 0){
+        //The 3rd and 4th test are redundant in the case of the
+        //same target levels as source levels
+        if(i > 1){ break;}
+        //In this case the target levels are the same as the source levels
+        n_layers_tgt[i] = n_layers_src[i];
+
+        printf ("      -> Testing %d source layers, %d target layers\n",
+                n_layers_src[i],n_layers_tgt[i]);
+      }
+      else{
+        printf ("      -> Testing %d source layers, %d target layers\n",
+                n_layers_src[i],n_layers_tgt[i]);
+      }
+      auto npacks_src = ekat::PackInfo<N>::num_packs(n_layers_src[i]);
+      auto npacks_tgt = ekat::PackInfo<N>::num_packs(n_layers_tgt[i]);
+      auto p_tgt = view_1d<Pack<Real,N>>("",npacks_tgt);
+      auto p_tgt_s = Kokkos::create_mirror_view(ekat::scalarize(p_tgt));
+      auto tmp_src = view_2d<Pack<Real,N>>("",2,npacks_src);
+      auto tmp_src_s = Kokkos::create_mirror_view(ekat::scalarize(tmp_src));
+      auto p_src = view_2d<Pack<Real,N>>("",2,npacks_src);
+      auto p_src_s = Kokkos::create_mirror_view(ekat::scalarize(p_src));
+      auto out = view_2d<Pack<Real,N>>("",2,npacks_tgt);
+      auto out_s = Kokkos::create_mirror_view(ekat::scalarize(out));
+      auto mask = view_2d<Mask<N>>("",2,npacks_tgt);
+  
+      //Set target levels    
+      for (int lev=0; lev<(n_layers_tgt[i]-1); lev++){
+        p_tgt_s(lev) = 2*lev+perturb;
+      } 
+      //Set source levels and source input (tmp_src)
+      for (int col=0; col<2; col++){
+        for (int lev=0; lev<n_layers_src[i]; lev++){
+          p_src_s(col,lev) = 2.*lev;
+          tmp_src_s(col,lev) = 100.*(col+1) + 2.*lev;
+        }//end of looping over levels
+      }//end of looping over columns
+  
+      perform_vertical_interpolation(p_src,
+                                     p_tgt,
+                                     tmp_src,
+                                     out,
+                                     mask,
+                                     n_layers_src[i],
+                                     n_layers_tgt[i]);
+
+      //Check that output of interpolation is as expected
+      for(int col=0; col<2; col++){
```

Comment:
These checks won't work on GPU, since `out_s` does not contain the updated output. You need a deep_copy back to host in order to read the correct results.

Reply:
Added the deep copy back